src/openms/source/CHEMISTRY/Enzyme.cpp

```diff
@@ -0,0 +1,236 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2015.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Xiao Liang  $
+// $Authors: Xiao Liang $
+// --------------------------------------------------------------------------
+//
+
+#include <OpenMS/CHEMISTRY/Enzyme.h>
+#include <cstdlib>
+#include <iostream>
+
+using namespace std;
+
+namespace OpenMS
+{
+  // enzyme
+  Enzyme::Enzyme() :
+    name_("unknown_enzyme"),
+    cleavage_regex_(""),
+    regex_description_(""),
+    n_term_gain_(""),
+    c_term_gain_(""),
+    psi_id_(""),
+    xtandem_id_(""),
+    omssa_id_()
+  {
+  }
+
+  Enzyme::Enzyme(const Enzyme & enzyme) :
+    name_(enzyme.name_),
+    synonyms_(enzyme.synonyms_),
+    cleavage_regex_(enzyme.cleavage_regex_),
+    regex_description_(enzyme.regex_description_),
+    n_term_gain_(enzyme.n_term_gain_),
+    c_term_gain_(enzyme.c_term_gain_),
+    psi_id_(enzyme.psi_id_),
+    xtandem_id_(enzyme.xtandem_id_),
+    omssa_id_(enzyme.omssa_id_)
+  {
+  }
+
+  Enzyme::Enzyme(const String & name,
```

Comment:
After changing the constructors, I've not yet tested PyOpenMS. report this later